CorvallisBus.Core/TransitManager.cs

```diff
@@ -61,18 +61,41 @@ public static async Task<ClientBusSchedule> GetSchedule(ITransitRepository repos
                         currentTime));
         }
 
+        private static BusStopRouteDaySchedule? GetBestGuessDaySchedule(BusStopRouteSchedule routeSchedule, DateTimeOffset currentTime)
+        {
+            BusStopRouteDaySchedule? daySchedule = null;
+            var potentialDaySchedules = routeSchedule.DaySchedules.Where(ds => DaysOfWeekUtils.TodayMayFallInsideDaySchedule(ds, currentTime));
+            if (potentialDaySchedules.Count() == 1) daySchedule = potentialDaySchedules.First();
```

Comment:
avoid multiple Count(), it is doing multiple enumeration of the lazy enumerable returned by .Where